x-pack/plugins/security_solution/cypress/e2e/detection_alerts/ransomware_alerts_display.cy.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setStartDate } from '../../tasks/date_picker';
+import { waitForAlertsToPopulate } from '../../tasks/create_new_rule';
+import { login, visit, visitWithoutDateRange } from '../../tasks/login';
+
+import { ALERTS_URL, TIMELINES_URL } from '../../urls/navigation';
+import { ALERT_DATA_GRID, SELECT_HISTOGRAM, TREND_CHART_LEGEND } from '../../screens/alerts';
+import { esArchiverLoad, esArchiverUnload } from '../../tasks/es_archiver';
+import { TIMELINE_QUERY } from '../../screens/timeline';
+
+describe('Ransomware alerts display', () => {
+  before(() => {
+    login();
+    esArchiverLoad('ransomware');
+  });
+
+  after(() => {
+    esArchiverUnload('ransomware');
+  });
+
+  describe('Ransomware display in Alerts Section', () => {
+    beforeEach(() => {
+      visitWithoutDateRange(ALERTS_URL);
+      const dateContainingAllEvents = 'Jul 27, 2015 @ 00:00:00.000';
+      setStartDate(dateContainingAllEvents);
+      waitForAlertsToPopulate();
+    });
+
+    describe('Alerts table', () => {
+      it('shows Ransomware Alerts', () => {
+        cy.get(ALERT_DATA_GRID)
+          .invoke('text')
+          .then((text) => {
+            expect(text).contains('Ransomware Prevention Alert');
```

Comment:
When performing assertions on texts, it is better to do not use `contains` when possible since this is not matching the string literally. If the text was for example `Foo Ransomware Prevention Alert` this test would pass without us being aware of the new introduced word.

The best way of doing this to avoid bypassing issues is to grab the element that has the value we want to assert and check that the it has the exact text we are expecting.

`cy.get(ELEMENT_THAT_HAS_THE_TEXT).should('have.text', 'expectedText' )`

Reply:
This way should be used on the different assertions we are performing in this file.